include/matrix3x4SIMD_impl.h

```diff
@@ -217,9 +217,9 @@ inline matrix3x4SIMD& matrix3x4SIMD::setScale(const core::vectorSIMDf& _scale)
 	vectorSIMDu32& rowAlias0 = reinterpret_cast<vectorSIMDu32&>(rows[0]);
 	vectorSIMDu32& rowAlias1 = reinterpret_cast<vectorSIMDu32&>(rows[1]);
 	vectorSIMDu32& rowAlias2 = reinterpret_cast<vectorSIMDu32&>(rows[2]);
-	rowAlias0 = (scaleAlias & BUILD_MASKF(1, 0, 0, 0)) | (rowAlias0 & mask0001);
-	rowAlias1 = (scaleAlias & BUILD_MASKF(0, 1, 0, 0)) | (rowAlias1 & mask0001);
-	rowAlias2 = (scaleAlias & BUILD_MASKF(0, 0, 1, 0)) | (rowAlias2 & mask0001);
+	rowAlias0 = (scaleAlias & static_cast<vectorSIMDf>(BUILD_MASKF(1, 0, 0, 0))) | (rowAlias0 & static_cast<vectorSIMDf>(mask0001));
+	rowAlias1 = (scaleAlias & static_cast<vectorSIMDf>(BUILD_MASKF(0, 1, 0, 0))) | (rowAlias1 & static_cast<vectorSIMDf>(mask0001));
+	rowAlias2 = (scaleAlias & static_cast<vectorSIMDf>(BUILD_MASKF(0, 0, 1, 0))) | (rowAlias2 & static_cast<vectorSIMDf>(mask0001));
```

Comment:
vectorSIMDf constructor from __m128 is explicit, so use explicit constructor call `vectorSIMDf(m128_value)` instead of `static_cast<vectorSIMDf>(m128_value)`.

`static_cast` is only what you want to see for PoD to PoD conversions and pointer conversions between related classes.

Explicit constructor invocation `T(u)` is not the same as a C-style cast `(T)u`